numbat/src/interpreter.rs

```diff
@@ -30,8 +30,17 @@ pub enum RuntimeError {
     AssertFailed(Span),
     #[error("Assertion failed because the following two values are not the same:\n  {1}\n  {3}")]
     AssertEq2Failed(Span, Value, Span, Value),
-    #[error("Assertion failed because the following two quantities differ by more than {4}:\n  {1}\n  {3}")]
-    AssertEq3Failed(Span, Quantity, Span, Quantity, Quantity),
+    #[error("Assertion failed because the following two quantities differ by {7}, which is more than or equal to {6}:\n  {2} ({1})\n  {5} ({4})")]
+    AssertEq3Failed(
+        Span,
+        Quantity,
+        Quantity,
+        Span,
+        Quantity,
+        Quantity,
+        Quantity,
+        Quantity,
```

Comment:
Maybe turn this into a struct (`AssertEq3Error`)? We could then implement `Display` for that struct and use `#[error("{0}")]` here instead.

This would allow us to have a bit more control over how the message looks like. For example, this would allow us to prevent the duplication in a (pretty standard) case like this:
```
>>> assert_eq(200 cm, 202 cm, 1 cm)
error: Assertion failed
  ┌─ <input:10>:1:19
  │
1 │ assert_eq(200 cm, 202 cm, 1 cm)
  │           ------  ^^^^^^ 202 cm (202 cm)
  │           │        
  │           200 cm (200 cm)
  │
  = Assertion failed because the following two quantities differ by 2 cm, which is more than or equal to 1 cm:
      200 cm (200 cm)
      202 cm (202 cm)
```

Reply:
A struct sounds good. To check my understanding, do you mean to reduce the noise from printing the lhs and rhs quantities in their original units in parentheses when the units are the same?

Something like this?:
```
>>> assert_eq(2000mm, 201cm, 1mm)
error: Assertion failed
  ┌─ <input:3>:1:19
  │
1 │ assert_eq(2000mm, 201cm, 1mm)
  │           ------  ^^^^^ 2010.0 mm (201 cm)
  │           │
  │           2000 mm
  │
  = Assertion failed because the following two quantities differ by 10.0 mm, which is more than or equal to 1 mm:
      2000 mm

      201 cm (2010.0 mm)
```

Reply:
Exactly.

I think I would also "more than" instead of "more than or equal". Any change the actual logic accordingly.